Web test config support in run settings

ganesp opened the pull request:
## Description
Updating the TestPlatformExternalsVersion to 15.7.0-preview-1409276 for the following changes:
1)
DEVDIV Pull Request [105172](https://devdiv.visualstudio.com/DevDiv/VS.in%20Agile%20Testing%20IDE/_git/VS/pullrequest/105172?_a=overview): Web test configuration support in run settings
DEVDIV Bug [569865](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/569865): Web test configuration support in run settings 


## Related issue
NA

ganesp committed:
Updated the TestPlatformExternalsVersion to 15.6.0-preview-1406082

Updated the TestPlatformExternalsVersion to 15.6.0-preview-1406082
Changes: Added web test config support in run settings

ganesp committed:
Revert "Updated the TestPlatformExternalsVersion to 15.6.0-preview-1406082"

This reverts commit 756a236b1d9e0ee5b18a6e87210d0c30ccef2928.

ganesp committed:
Web test config support in run settings

Web test config support in run settings
Updating the TestPlatformExternalsVersion to 15.6.0-preview-1406082

smadala commented:
@ganesp Add payload changes PR(DevDiv PR)  and any devdiv bug/user story to description .

ganesp commented:
@smadala Done, updated the description.
@singhsarab Can you please approve and merge the PR?

ganesp committed:
Added acceptance test for web test run config in run settings scenario

The test WebTestRunAllTestsWithRunSettings validates the
simulateThinkTimes setting specified in the WebTestRunConfiguration in
run settings.

ganesp committed:
Minor change to remove an unused variable in test code

Minor change to remove an unused variable in test code

smadala commented on scripts/build/TestPlatform.Dependencies.props:
```diff
@@ -25,6 +25,6 @@
     <JsonNetVersion>9.0.1</JsonNetVersion>
     <MoqVersion>4.7.63</MoqVersion>
-    <TestPlatformExternalsVersion>15.6.0-preview-1329213</TestPlatformExternalsVersion>
+    <TestPlatformExternalsVersion>15.6.0-preview-1406082</TestPlatformExternalsVersion>
```
Change this version to 15.7 for consistency.

ganesp committed:
Replaced test WebTestRunAllTests with WebTestRunAllTestsWithRunSettings

Replaced test WebTestRunAllTests with WebTestRunAllTestsWithRunSettings

The latter test has added validation for web test config in run settings

smadala commented on test/Microsoft.TestPlatform.TestUtilities/IntegrationTestBase.cs:
```diff
@@ -145,7 +145,7 @@ public void ExecuteNotSupportedRunnerFrameworkTests(string runnerFramework, stri
         /// <param name="failedTestsCount">Failed test count</param>
         /// <param name="skippedTestsCount">Skipped test count</param>
-        public void ValidateSummaryStatus(int passedTestsCount, int failedTestsCount, int skippedTestsCount)
+        public void ValidateSummaryStatus(int passedTestsCount, int failedTestsCount, int skippedTestsCount, int minExecutionTimeInSeconds=0)
```
Update `param` comment.

ganesp replied:
Done.

ganesp committed:
Updated the CreateRunSettingsFile to accept the run settings xml

Updated the CreateRunSettingsFile to accept the run settings xml instead
of run settings node

ganesp committed:
Updated TestPlatformExternalsVersion to 15.7.0-preview-1409276

Updated TestPlatformExternalsVersion to 15.7.0-preview-1409276

ganesp committed:
Added param comment for ValidateSummaryStatus

Added param comment for ValidateSummaryStatus

smadala commented:
@dotnet-bot test Ubuntu14.04 / Debug Build.

ganesp commented:
@dotnet-bot test this.

singhsarab commented on test/Microsoft.TestPlatform.AcceptanceTests/DifferentTestFrameworkSimpleTests.cs:
```diff
@@ -45,10 +45,25 @@ public void CPPRunAllTestExecutionPlatformx64(RunnerInfo runnerInfo)
+
+            //minExecutionTimeInSeconds is set to 30 here as the web test has a request with think time equal to 30 seconds
+            //therefore, the test will at least run for 30 seconds
+            WebTestRunAllTests(runnerInfo.RunnerFramework, runSettingsFilePath, 30);
```
Can we reduce this please.
This will simply add 30 sec to the overall run

ganesp replied:
I discussed this at length with Satya. It is added for a purpose. Also, we are replacing the old test with this one so gaining few seconds from the old test.

ganesp replied:
Using iterationCount/web test result file size to validate the web test config.
Removed this think time logic.

singhsarab commented on test/Microsoft.TestPlatform.TestUtilities/IntegrationTestBase.cs:
```diff
@@ -508,6 +520,24 @@ public static void CreateRunSettingsFile(string destinationRunsettingsPath, IDic
+        /// <param name="runSettingsXml">
+        /// Xml string
+        /// </param>
+        public static void CreateRunSettingsFile(string destinationRunsettingsPath, string runSettingsXml)
```
Don't we already have this ?

ganesp commented:
@dotnet-bot test Ubuntu14.04 / Debug Build.

ganesp commented:
@dotnet-bot test Windows_NT / Release Build.

ganesp commented:
@dotnet-bot test Windows_NT / Debug Build.

ganesp committed:
Merge branch 'master' into webtestconfig

ganesp committed:
Updated the test WebTestRunAllTestsWithRunSettings to use iterationCount for validation

Updated the test WebTestRunAllTestsWithRunSettings to use iterationCount
for validation

ganesp committed:
Updated a comment for WebTestRunAllTestsWithRunSettings

Updated a comment for WebTestRunAllTestsWithRunSettings

ganesp committed:
Merge branch 'webtestconfig' of https://github.com/ganesp/vstest into webtestconfig

ganesp committed:
Minor comment updates

Minor comment updates

ganesp committed:
Fixed the test WebTestRunAllTestsWithRunSettings

Fixed the test WebTestRunAllTestsWithRunSettings

ganesp committed:
Fixed the test WebTestRunAllTestsWithRunSettings

Fixed the test WebTestRunAllTestsWithRunSettings
